tests/unit/websocket_test.cc

```diff
@@ -64,3 +81,76 @@ SEASTAR_TEST_CASE(test_websocket_handshake) {
         }
     });
 }
+
+
+
+SEASTAR_TEST_CASE(test_websocket_handler_registration) {
+    return seastar::async([] {
+        loopback_connection_factory factory;
+        loopback_socket_impl lsi(factory);
+
+        auto acceptor = factory.get_server_socket().accept();
+        auto connector = lsi.connect(socket_address(), socket_address());
+        connected_socket sock = connector.get0();
+        auto input = sock.input();
+        auto output = sock.output();
+
+        // Setup server
+        websocket::server ws;
+        ws.register_handler("echo", [] (input_stream<char>& in,
+                        output_stream<char>& out) {
+            return repeat([&in, &out]() {
+                return in.read().then([&out](temporary_buffer<char> f) {
+                    std::cerr << "f.size(): " << f.size() << "\n";
+                    if (f.empty()) {
+                        return make_ready_future<stop_iteration>(stop_iteration::yes);
+                    } else {
+                        return out.write(std::move(f)).then([&out]() {
+                            return out.flush().then([] {
+                                return make_ready_future<stop_iteration>(stop_iteration::no);
+                            });
+                        });
+                    }
+                });
+            });
+        });
+        websocket::connection conn(ws, acceptor.get0().connection);
+        future<> serve = conn.process();
+
+        auto close = defer([&conn, &input, &output, &serve] () noexcept {
+            conn.shutdown();
+            conn.close().get();
+            input.close().get();
+            output.close().get();
+            serve.get();
+         });
+
+        // handshake
+        const std::string request =
+                "GET / HTTP/1.1\r\n"
+                "Upgrade: websocket\r\n"
+                "Connection: Upgrade\r\n"
+                "Sec-WebSocket-Key: dGhlIHNhbXBsZSBub25jZQ==\r\n"
+                "Sec-WebSocket-Version: 13\r\n"
+                "Sec-WebSocket-Protocol: echo\r\n"
+                "\r\n";
+        output.write(request).get();
+        output.flush().get();
+        input.read().get();
```

Comment:
same - we need to make sure to read until eof (or until we read everything we wanted, as the previous test case did with a consumer)